### ISSUE TITLE
Should reject using / await using in a switch case / default clause

### ISSUE THREAD

**Issue by sapphi-red** [microsoft/TypeScript#62708]
### 🔎 Search Terms

explicit resource management, using, await using, switch, case clause, default clause

### 🕗 Version & Regression Information

- This changed between versions ______ and _______
- This changed in commit or PR _______
- This is the behavior in every version I tried, and I reviewed the FAQ for entries about _________
- I was unable to test this on prior versions because _______

### ⏯ Playground Link

https://www.typescriptlang.org/play/?target=99&ts=6.0.0-dev.20251030#code/M4dwlgLgxgFgBACgLIEMIwHQCcUDsAmA9gLYICUZcA3gFBxxQrACmcADAFx31wCuwYXAHM4+AExs4AXmpwA2gGUAnsQBGhADYZ8YYAAdCLALrlqAXzhmA3N3qqszFAGsb3RizgBGLjzgoQKJB8AsKiYp7Ssooq6lo6+obMJpRUFta2cPaOLjRmNDTMAB4GWBDmNEA

### 💻 Code

```ts
switch (Math.random()) {
  case 0:
    using d20 = { [Symbol.dispose]() {} };
    break;

  case 1:
    await using d21 = { [Symbol.dispose]() {} };
    break;
}

export {}
```

### 🙁 Actual behavior

No error happens.

### 🙂 Expected behavior

The error is reported that `using` / `await using` cannot be used here.

### Additional information about the issue

The spec was updated to disallow this (https://github.com/rbuckton/ecma262/pull/14).
Related PRs in other projects:
- https://github.com/babel/babel/pull/17323
- https://github.com/acornjs/acorn/pull/1374

Some tests needs to be updated. For example: https://github.com/microsoft/TypeScript/blob/1654ebfb54facadb3a0468ced0ca779ee323c7e5/tests/cases/conformance/statements/VariableStatements/usingDeclarations/usingDeclarations.1.ts#L98-L113

**Comment by RyanCavanaugh** [comment]
🤖 This is an automated response. I looked for things that might help (duplicates, FAQ entries, etc) but didn't find anything. A human will take a look!
